react/states_and_effects/more_on_state.md

```diff
@@ -16,11 +16,11 @@ Managing and structuring state effectively is by far one of the most crucial par
 
 The assignment items go through the topic thoroughly, but as a general rule of thumb: don't put values in state that can be calculated using existing values, state, and/or props. 
 
-#### State is immutable
+#### State should not be mutated
 
-<span id="state-is-immutable"></span>Mutating state is no-go area in React as it leads to unpredictable results. Primitives are already immutable, but if you are using reference type values i.e. arrays and objects, never mutate them. Instead always use the `setState` function to update state. 
+Mutating state is a no-go area in React as it leads to unpredictable results. Primitives are already immutable, but if you are using reference type values i.e. arrays and objects, never mutate them. According to React documentation, we should treat state as if it was _immutable_. For more information about state mutating check [Why is mutating state not recommended in React?](https://react.dev/learn/updating-objects-in-state#why-is-mutating-state-not-recommended-in-react)
```

Comment:
We try to have all the readings in the assignment or additional resources section of the lesson. Can we move this article to one of those sections please?

```suggestion
Mutating state is a no-go area in React as it leads to unpredictable results. Primitives are already immutable, but if you are using reference type values i.e. arrays and objects, never mutate them. According to React documentation, we should treat state as if it was _immutable_.
```